mp4/file_test.go

```diff
@@ -156,3 +157,64 @@ func TestDecodeEncode(t *testing.T) {
 		}
 	}
 }
+
+func TestFilesWithEmsg(t *testing.T) {
+	// File with ftyp, moov, styp, emsg, emsg, moof, mdat, moof, mdat
+	init := CreateEmptyInit()
+	init.AddEmptyTrack(uint32(48000), "audio", "en")
+	trak := init.Moov.Trak
+	err := trak.SetAACDescriptor(aac.AAClc, 48000)
+	if err != nil {
+		t.Error(err)
+	}
+	data := make([]byte, 0, init.Size())
+	buf := bytes.NewBuffer(data)
+	err = init.Encode(buf)
+	if err != nil {
+		t.Error(err)
+	}
+	seg := NewMediaSegment()
+	frag, err := CreateFragment(1, 1)
+	if err != nil {
+		t.Error(err)
+	}
+	frag.AddFullSample(FullSample{
+		Sample: Sample{
+			Flags:                 0x0,
+			Dur:                   1024,
+			Size:                  6,
+			CompositionTimeOffset: 0,
+		},
+		DecodeTime: 0,
+		Data:       []byte{0, 1, 2, 3, 4, 5},
+	})
+	emsg1 := EmsgBox{ID: 1}
+	frag.AddEmsg(&emsg1)
+	emsg2 := EmsgBox{ID: 2}
+	frag.AddEmsg(&emsg2)
```

Comment:
I think we can do this
```suggestion
	frag.AddEmsg(&EmsgBox{ID: 1})
	frag.AddEmsg(&EmsgBox{ID: 2})
```

Reply:
Yes, that can be simplified.